core/src/main/java/io/questdb/griffin/FunctionParser.java

```diff
@@ -530,9 +527,29 @@ private Function createFunction(
 
                 int sigArgTypeSum = 0;
                 for (int k = 0; k < sigArgCount; k++) {
-                    final Function arg = args.getQuick(k);
-                    final boolean undefined = arg.isUndefined();
                     final int sigArgTypeMask = descriptor.getArgTypeMask(k);
+                    final int sigArgType = FunctionFactoryDescriptor.toType(sigArgTypeMask);
+
+                    final Function arg = args.getQuick(k);
+                    final boolean argIsNull = arg.getType() == ColumnType.NULL;
```

Comment:
I'd try to use ColumnType.overloadPriority for NULL type to indicate that it can match any type

Reply:
thank you, I think this is a great idea, I am going to have a go at it.